aws/resource_aws_api_gateway_method_settings.go

```diff
@@ -113,16 +123,20 @@ func resourceAwsApiGatewayMethodSettingsRead(d *schema.ResourceData, meta interf
 		return nil
 	}
 
-	d.Set("settings.0.metrics_enabled", settings.MetricsEnabled)
-	d.Set("settings.0.logging_level", settings.LoggingLevel)
-	d.Set("settings.0.data_trace_enabled", settings.DataTraceEnabled)
-	d.Set("settings.0.throttling_burst_limit", settings.ThrottlingBurstLimit)
-	d.Set("settings.0.throttling_rate_limit", settings.ThrottlingRateLimit)
-	d.Set("settings.0.caching_enabled", settings.CachingEnabled)
-	d.Set("settings.0.cache_ttl_in_seconds", settings.CacheTtlInSeconds)
-	d.Set("settings.0.cache_data_encrypted", settings.CacheDataEncrypted)
-	d.Set("settings.0.require_authorization_for_cache_control", settings.RequireAuthorizationForCacheControl)
-	d.Set("settings.0.unauthorized_cache_control_header_strategy", settings.UnauthorizedCacheControlHeaderStrategy)
+	if err := d.Set("settings", []interface{}{map[string]interface{}{
```

Comment:
ah this could be a good use case of using an `expander function pattern` as mentioned in slack to set the "settings" attribute